src/main/java/uk/gov/hmcts/juror/api/moj/repository/letter/court/DeferralGrantedLetterListRepositoryImpl.java

```diff
@@ -1,88 +1,159 @@
+                .firstName(tuple.get(QJuror.juror.firstName))
+                .lastName(tuple.get(QJuror.juror.lastName))
+                .postcode(tuple.get(QJuror.juror.postcode))
+                .status(tuple.get(QJurorStatus.jurorStatus.statusDesc))
+                .deferralDate(tuple.get(QJurorPool.jurorPool.deferralDate))
+                .deferralReason(tuple.get(EXCUSAL_CODE.excusalCode.description))
+                .datePrinted(tuple.get(QJurorHistory.jurorHistory.dateCreated))
+                .isActive(tuple.get(QJurorPool.jurorPool.isActive))
+                .build())
+            .collect(Collectors.toList());
+
+        // for each juror select the most recent deferred date and remove the other entries
+        // this is to ensure that only the most recent deferral date is returned
+        ConcurrentHashMap<String, DeferralGrantedLetterList> deferralGrantedLetterListMap = new ConcurrentHashMap<>();
+        deferralGrantedLetterLists.forEach(deferralGrantedLetterList -> {
+            String jurorNumber = deferralGrantedLetterList.getJurorNumber();
+            if (deferralGrantedLetterListMap.containsKey(jurorNumber)) {
+                DeferralGrantedLetterList existingDeferralGrantedLetterList =
+                    deferralGrantedLetterListMap.get(jurorNumber);
+                if (deferralGrantedLetterList.getDeferralDate().isAfter(existingDeferralGrantedLetterList.getDeferralDate())) {
+                    deferralGrantedLetterListMap.put(jurorNumber, deferralGrantedLetterList);
+                }
+            } else {
+                deferralGrantedLetterListMap.put(jurorNumber, deferralGrantedLetterList);
+            }
+        });
+
+        return deferralGrantedLetterListMap.values().stream().collect(Collectors.toList());
     }
 
 
-    private void filterEligibleLetterSearchCriteria(JPAQuery<DeferralGrantedLetterList> jpaQuery,
+    private JPAQuery<Tuple> buildBaseQuery(String locCode) {
+
+        JPAQueryFactory queryFactory = new JPAQueryFactory(entityManager);
+
+        return queryFactory.select(
+                QJurorPool.jurorPool.pool.poolNumber,
+                QJuror.juror.jurorNumber,
+                QJuror.juror.firstName,
+                QJuror.juror.lastName,
+                QJuror.juror.postcode,
+                QJurorPool.jurorPool.deferralDate,
+                QJurorStatus.jurorStatus.statusDesc,
+                QJurorHistory.jurorHistory.dateCreated,
+                QJurorPool.jurorPool.isActive,
+                EXCUSAL_CODE.excusalCode.description)
+            .from(QJurorPool.jurorPool)
+            .join(QJuror.juror).on(QJuror.juror.jurorNumber.eq(QJurorPool.jurorPool.juror.jurorNumber))
+            .join(QPoolRequest.poolRequest)
+            .on(QPoolRequest.poolRequest.poolNumber.eq(QJurorPool.jurorPool.pool.poolNumber))
+            .join(QJurorStatus.jurorStatus).on(QJurorStatus.jurorStatus.eq(QJurorPool.jurorPool.status))
```

Comment:
Should not be needed as we can just user QjurorPool.jurorPool.juror & QjurorPool.jurorPool.pool & QjurorPool.jurorPool.status

Reply:
have tried without the explicit joins but get the error as below
  : ERROR: missing FROM-clause entry for table "j1_0"\n  Position: 488 
indicating the underlying query doesnt generate correctly without it

Reply:
I believe this is because you are still searching for QJuror instead of QJurorPool here
<img width="592" alt="image" src="https://github.com/user-attachments/assets/28333081-ab34-4ee1-aba8-410aa4dcf9dd">